src/hera/workflows/user_container.py

```diff
@@ -13,16 +13,82 @@
 from hera.workflows.resources import Resources
 from hera.workflows.volume import _BaseVolume
 
+# TODO: Mixins imports user container, so this avoids a circular import. Fix in a future release
+EnvT = Optional[
+    Union[
+        _BaseEnv,
+        EnvVar,
+        List[Union[_BaseEnv, EnvVar, Dict[str, Any]]],
+        Dict[str, Any],
+    ]
+]
+"""`EnvT` is the core Hera type for environment variables.
+
+The env type enables setting single valued environment variables, lists of environment variables, or dictionary 
+mappings of env variables names to values, which are automatically parsed by Hera.
+"""
+
+# TODO: Mixins imports user container, so this avoids a circular import. Fix in a future release
+EnvFromT = Optional[Union[_BaseEnvFrom, EnvFromSource, List[Union[_BaseEnvFrom, EnvFromSource]]]]
+"""`EnvFromT` is the core Hera type for environment variables derived from Argo/Kubernetes sources.
+
+This env type enables specifying environment variables in base form, as `hera.workflows.env` form, or lists of the 
+aforementioned objects.
+"""
+
 
 class UserContainer(_ModelUserContainer):
     """`UserContainer` is a container type that is specifically used as a side container."""
 
-    env: Optional[List[Union[_BaseEnv, EnvVar]]] = None  # type: ignore[assignment]
-    env_from: Optional[List[Union[_BaseEnvFrom, EnvFromSource]]] = None  # type: ignore[assignment]
+    env: EnvT = None  # type: ignore[assignment]
+    env_from: EnvFromT = None  # type: ignore[assignment]
```

Comment:
Why does the linter have an issue with assignment here? 🤔 Are the "ignores" actually still needed?

Reply:
> Why does the linter have an issue with assignment here? 🤔 Are the "ignores" actually still needed?

Copy / pasta, sorry, will change

Reply:
Parent class has these fields that are overwritten <img width="544" alt="Screenshot 2023-11-01 at 07 15 29" src="https://github.com/argoproj-labs/hera/assets/18152652/30edbd9a-7bc3-46db-9d92-b969eb8bf03c">

Reply:
And you get this nonsense `src/hera/workflows/user_container.py:43: error: Incompatible types in assignment (expression has type "Union[_BaseEnv, EnvVar, List[Union[_BaseEnv, EnvVar, Dict[str, Any]]], Dict[str, Any], None]", base class "UserContainer" defined the type as "Optional[List[EnvVar]]")  [assignment]`